### ISSUE TITLE
[Feature request] Support for BitBox02 hardware wallet

### ISSUE THREAD

**Issue by eddex** [Berry-Pool/nami-wallet#451]
Nami supports Ledger and Trezor hardware wallets, which is great. But there is a third player in the hardware wallet game that may be a little less known, but of equal significance as the other two.

So far [Adalite](https://adalite.io/) is the only Cardano wallet that supports BitBox02 hardware wallets. Unfortunately Adalite doesn't support Cardano dApps, so it can only be used to send funds to other wallets and for staking. The creators of Adalite are not going to implement dApp support. (see Adalite issue [#1280](https://github.com/vacuumlabs/adalite/issues/1280#issuecomment-1065249732)).

It would be great if Nami would be the first Cardano wallet that supports BitBox02 hardware wallets for dApps!

- An overview of technical resources can be found here: https://shiftcrypto.ch/dev/
- There is a JS library for connecting to the BitBox02 hardware wallet: https://github.com/digitalbitbox/bitbox02-api-js

**Comment by eddex** [comment]
started working on this here: https://github.com/eddex/nami-wallet/tree/feat/bitbox02-hw-wallet

**Comment by alessandrokonrad** [comment]
Hey, that's cool. You think you can get a full PR done? I heared of BitBox before, but I'm not very familiar with it. I don't have one, so I can't really test it. Does it already support plutus functionality?

**Comment by alessandrokonrad** [comment]
> Is there a difference in signing a simple tx compared to a SC tx? Or what do you mean by "plutus functionality"?

Yeah, with plutus functionality I mean having additional information in the tx, like datum, redeemer, the plutus script itself etc. Ledger and Trezor can't handle it yet for example.

**Comment by eddex** [comment]
> > Is there a difference in signing a simple tx compared to a SC tx? Or what do you mean by "plutus functionality"?
> 
> Yeah, with plutus functionality I mean having additional information in the tx, like datum, redeemer, the plutus script itself etc. Ledger and Trezor can't handle it yet for example.

Have to check that. But I'd assume not. Someone from @digitalbitbox can surely answer this.

**Comment by eddex** [comment]
It's not as straight forward to integrate as I expected because the flow of connecting a BitBox02 wallet is a bit different than Ledger or Trezor. I haven't been able to get it working yet. Everything I've done so far is in my branch.
